pkg/cloud/rgraph/rnode/all/new.go

```diff
@@ -18,40 +18,37 @@ package all
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
 	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/address"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/backendservice"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/fake"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/forwardingrule"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/healthcheck"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/networkendpointgroup"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/targethttpproxy"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/tcproute"
-	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/urlmap"
 )
 
+type newBuilderFunc func(id *cloud.ResourceID) rnode.Builder
+
+var (
+	builderFuncsLock sync.Mutex
+	builderFuncs     = map[string]newBuilderFunc{}
+)
+
+// RegisterBuilder constructor. shortName is the plural lowerCamelCase resource
```

Comment:
RegisterBuilder is not a constructor. The global variables are already created and this function is adding ("registering") the values so they can be used by rGraph. Can you change a bit the description?

Reply:
The sentence is supposed to be "Register Builder constructor." but this is not allowed by go.. I will clarify by expanding the description.